src/wipplpy/modules/connection.py

```diff
@@ -0,0 +1,118 @@
+            try:
+                self._local_connect(shot_number, tree_name)
+            except Exception as e:
+                # Local connection failed. Try a remote conn. for completeness
+                logging.debug(
+                    "Failed attempted local connection: `%s`\nTrying a"
+                    " remote connection.",
+                    e,
+                )
+                try:
+                    self._remote_connect(shot_number, tree_name, server_name)
+                except Exception as e:
+                    # Neither connection type succeeded. Raise the exception
+                    self.is_remote_connection = None  # reset the flag
+                    logging.exception(
+                        "An error occurred while attempting to connect to"
+                        " `%s`: `%s`",
+                        server_name,
+                        e,
+                    )
+                    raise
+
+        else:
+            try:
+                self._remote_connect(shot_number, tree_name, server_name)
+            except Exception as e:
+                # Remote connection failed. Try a local conn. for completeness
+                logging.debug(
+                    "Failed attempted remote connection: `%s`\nTrying a"
+                    " local connection.",
+                    e,
+                )
+                try:
+                    self._local_connect(shot_number, tree_name)
+                except Exception as e:
+                    # Neither connection type succeeded. Raise the exception
+                    self.is_remote_connection = None  # reset the flag
+                    logging.exception(
+                        "An error occurred while attempting to connect to"
+                        " `%s`: `%s`",
+                        server_name,
+                        e,
+                    )
+                    raise
+
+    def _local_connect(self, shot_number, tree_name):
+        """
+        Open a local MDSplus tree database.
+
+        Parameters
+        ----------
+        shot_number : `int`
+            The shot number from which to extract MDSplus data.
+        tree_name : `str`
+            String representing the tree name of the device's MDSplus database.
+        """
+        self.is_remote_connection = False
+
+        self.tree = mds.Tree(tree_name, shot_number)
+
+    def _remote_connect(self, shot_number, tree_name, server_name):
+        """
+        Open a remote MDSplus tree database.
+
+        Parameters
+        ----------
+        shot_number : `int`
+            The shot number from which to extract MDSplus data.
+        tree_name : `str`
+            String representing the tree name of the device's MDSplus database.
+        server_name : `str`
+            String representing the server in which the shot's data is located.
+        """
+        self.is_remote_connection = True
+
+        self.tree = mds.Connection(server_name)
```

Comment:
Add logging statement and in the logging statement about trying a remote connection maybe mention to the user something like "... If this is taking a long time, make sure you are using your WiscVPN static IP address.".